retriever/engines/mysql.py

```diff
@@ -39,6 +43,26 @@ class engine(Engine):
                       "{db}.{table}"),
                      ]
 
+    def check_credentials(self):
+        self.get_input()
+        try:
+            sql_conf = open_fr(MYSQL_CONF_PATH)
+            sql_conf_lines = sql_conf.readlines()
+            if self.opts["user"] != "root":
+                for line in sql_conf_lines:
```

Comment:
@jainamritanshu, do we really need this try block,  and why are we not considering ``root`` in this if
`if self.opts["user"] != "root"`

Could you give me a scenario in case I am under looking some details.

Regardless of the user value, we should be able to get the password set.
I am imagining a`` user`` as ``root`` with a password different from the default. Whatever the case may be, if we have two sets of values, the defaults values `required_opts` and the actual user supplied values when a command like 
`` retriever install mysql -u root -p password12! iris`` is used.


I think we should do something like this, 

```
# (so here we shall not need an else just process the file if it exist)
if os.path.exists(MYSQL_CONF_PATH): 
    (go ahead and read the values from line)
    for line in sql_conf_lines:
        (get the corresponding value of self.opts["user"])
        if there is a self.opts["user"] in line get the value of password. 
            if that value of password is not set in the conf file  exit the loop(we use defaults then)
            if the value of the password is there then we modify out dictionary.
            self.sql_lookup = {"user": username, "password": password} and exit the loop(no need to loop through other lines)
            by the end of the procedures, we shall have either set the new  sql_lookup tables or we use the default ones
```